### ISSUE TITLE
Suggestion selection does not work properly while using the keyboard buttons

### ISSUE THREAD

**Issue by vasilicatamas** [mozilla/addons#4100]
**Steps to reproduce:**
1.Navigate to https://addons-dev.allizom.org/en-US/firefox/
2.Type a key in the search bar.
3.Hover over any item from drop-down list using the keyboard arrows and press enter.

**Expected results:**
The corresponding add-on listing page is loaded.

**Actual results:**
The search is performed.

**Additional notes:**
- This issue reproduces on AMO-dev and AMO-stage using Firefox 54.0a1 (2017-03-01) under Windows 10 64-bit.
- Please see the attached screencast for this issue:
![animation](https://cloud.githubusercontent.com/assets/18114210/23510233/5ba5e548-ff61-11e6-9eb5-8c64a3cc5d22.gif)
